tests/helpers.py

```diff
@@ -0,0 +1,241 @@
+        address: str,
+        private_key: Optional[str],
+        lsig: Optional[transaction.LogicSig] = None,
+    ):
+        self.address = address
+        self.private_key = private_key
+        self.lsig = lsig
+
+        assert self.private_key or self.lsig
+
+    def mnemonic(self) -> str:
+        return mnemonic.from_private_key(self.private_key)
+
+    def is_lsig(self) -> bool:
+        return bool(not self.private_key and self.lsig)
+
+    @classmethod
+    def create(cls) -> "Account":
+        private_key, address = account.generate_account()
+        return cls(private_key=private_key, address=str(address))
+
+    @property
+    def decoded_address(self):
+        return encoding.decode_address(self.address)
+
+
+def get_kmd_accounts(
+    kmd_wallet_name="unencrypted-default-wallet", kmd_wallet_password=""
+):
+    kmd_client = _kmd_client()
+    wallets = kmd_client.list_wallets()
+
+    walletID = None
+    for wallet in wallets:
+        if wallet["name"] == kmd_wallet_name:
+            walletID = wallet["id"]
+            break
+
+    if walletID is None:
+        raise Exception("Wallet not found: {}".format(kmd_wallet_name))
+
+    walletHandle = kmd_client.init_wallet_handle(walletID, kmd_wallet_password)
+
+    try:
+        addresses = kmd_client.list_keys(walletHandle)
+
+        privateKeys = [
+            kmd_client.export_key(walletHandle, kmd_wallet_password, addr)
+            for addr in addresses
+        ]
+
+        kmdAccounts = [
+            Account(address=addresses[i], private_key=privateKeys[i])
+            for i in range(len(privateKeys))
+        ]
+    finally:
+        kmd_client.release_wallet_handle(walletHandle)
+
+    return kmdAccounts
+
+
+## Teal Helpers
+
+
+def logged_bytes(b: str):
+    return bytes(b, "ascii").hex()
+
+
+def logged_int(i: int):
+    return i.to_bytes(8, "big").hex()
+
+
+def assert_stateful_output(expr: Expr, output: List[str]):
+    assert expr is not None
+
+    src = compile_stateful_app(expr)
+    assert len(src) > 0
+
+    compiled = fully_compile(src)
+    assert len(compiled["hash"]) == 58
+
+    app_id = create_app(
+        compiled["result"],
+        transaction.StateSchema(0, 16),
+        transaction.StateSchema(0, 64),
+    )
+
+    result = call_app(app_id)
+
+    assert result == output
+
+
+def assert_output(expr: Expr, output: List[str], **kwargs):
+    assert expr is not None
+
+    src = compile_app(expr)
+    assert len(src) > 0
+
+    compiled = fully_compile(src)
+    assert len(compiled["hash"]) == 58
+
+    result = execute_app(compiled["result"], **kwargs)
+    assert result == output
+
+
+def compile_app(method: Expr, version: int = 5):
+    return compileTeal(Seq(method, Int(1)), mode=Mode.Application, version=version)
+
+
+def compile_stateful_app(method: Expr, version: int = 5):
+    expr = Cond([Txn.application_id() == Int(0), Int(1)], [Int(1), Seq(method, Int(1))])
+    return compileTeal(expr, mode=Mode.Application, version=version)
```

Comment:
Can we just use the previous helper? Also: I'm in favour of `app` and `sig` instead of `stateful` and `stateless`.

Reply:
Since we need to do this in multiple stages so that the account can opt in, we need to handle the create method separately and not execute anything.

Good point on app/sig vs stateful/stateless

Reply:
See: https://github.com/algorand/pyteal-utils/pull/13#discussion_r775928613